R/FilteredData.R

```diff
@@ -586,6 +590,36 @@ FilteredData <- R6::R6Class( # nolint
     restore_state_from_bookmark = function(state, check_data_hash = TRUE) {
       stop("Pure virtual method")
     },
+    #' @description 
+    #' Disable the filter panel by adding `disable` attribute to `filter_add_vars` and `filter_active_vars ` id in the UI element.
+    #' In addition, it will store the existing filter states in a private field called `cached_states` 
+    #' before removing all filter states from the object.
+    filter_panel_disable = function() {
+      private$filter_turn <- FALSE
+      shinyjs::disable("filter_add_vars")
+      shinyjs::disable("filter_active_vars")
+      private$cached_states <- self$get_filter_state()
+      self$remove_all_filter_states()
+      invisible(NULL)
+    },
+    #' @description enable the filter panel
```

Comment:
```suggestion
    #' @description enable the filter panel
        #' Enable the filter panel by adding `enable` class to `filter_add_vars` and `filter_active_vars ` tags in the UI.
    #' In addition, it will restore the filter states from a private field called `cached_states`.
```